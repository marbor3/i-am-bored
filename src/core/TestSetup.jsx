import React from "react";
import { render } from "@testing-library/react";
import { StateProvider } from "../state/DashboardState";
import dashboardReducer, {
  createDashboardIniitalState,
} from "../state/DashboardReducer";

const mockGeolocation = {
  getCurrentPosition: jest.fn().mockImplementation((success) =>
    Promise.resolve(
      success({
        coords: {
          latitude: 51.14,
          longitude: 17.04,
        },
      })
    )
  ),
  watchPosition: jest.fn(),
};

global.navigator.geolocation = mockGeolocation;

const renderWithContext = (component, options) =>
  render(
    <StateProvider
      reducer={dashboardReducer}
      initialState={createDashboardIniitalState(options)}
    >
      {component}
    </StateProvider>
  );

export { renderWithContext };
