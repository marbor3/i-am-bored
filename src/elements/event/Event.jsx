import react, { useCallback, useEffect, useState } from 'react';
import Link from 'next/link';
import { ACTION_TOGGLE_FAV } from '../../state/DashboardReducer';
import { useDashboardDispatch } from '../../state/DashboardState';
import styles from './Event.module.css';

export default function Event({ name, dates, description, id, isFavourite }) {
  const dispatch = useDashboardDispatch();
  const toggleFav = useCallback(() => {
    dispatch({
      type: ACTION_TOGGLE_FAV,
      payload: {
        id,
      },
    });
  }, []);
  return (
    <article className={`${styles.article} ${isFavourite ? styles['article--fav'] : null}`} data-testid="event">
      <h2>
        <button onClick={toggleFav}>{isFavourite ? 'unfav' : 'fav'}</button>
        {name}
      </h2>
      {dates ? (
        <div>
          {dates.start.localDate} {dates.start.localTime}
        </div>
      ) : null}

      <main>{description}</main>
      <Link href={`/event/${encodeURIComponent(id)}`}>more info</Link>
    </article>
  );
}
