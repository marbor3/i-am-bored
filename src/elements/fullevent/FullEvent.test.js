import React from "react";
import { render } from "@testing-library/react";
import Event from "./FullEvent";
import { renderWithContext } from "../../core/TestSetup";

describe("Elements - Event", () => {
  test("renders given event", async () => {
    const eventTitle = "Steve Hackett Genesis Revisited - Seconds Out & More";
    const { getByTestId, container } = renderWithContext(
      <Event name={eventTitle} />
    );

    expect(getByTestId("event")).toHaveTextContent(eventTitle);
    expect(container).toMatchSnapshot();
  });
});
