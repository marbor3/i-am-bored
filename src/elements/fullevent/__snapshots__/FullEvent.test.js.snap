// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements - Event renders given event 1`] = `
<div>
  <article
    class="article"
    data-testid="event"
  >
    <h1>
      <button>
        fav
      </button>
      Steve Hackett Genesis Revisited - Seconds Out & More
    </h1>
    <main />
  </article>
</div>
`;
